crates/nu-utils/standard_library/tests.nu

```diff
@@ -63,8 +63,76 @@ def test_path_add [] {
     }
 }
 
+
+def test_dirs [] {
+
+    def "myassert eq" [
+        left:any 
+        right:any 
+        msg?:string = "..."
+        --verbose:bool = false (-v)  # enable to see successful tests and values of unequal tests
+    ] {
+        if $left != $right {
+            let start_span = (metadata $left).span.start
+            let end_span = (metadata $right).span.end
+            let fail_msg = (if $verbose {
+                                $"\n    left: ($left|debug)\nnot equal to\n   right: ($right|debug)"
+                            } else {""})
+            error make {msg: $"Assertion failed checking ($msg)($fail_msg)", label: {text: "Values not equal" start: $start_span end: $end_span}}
+        } else {
+            if $verbose {
+                echo $"check succeeded: ($msg)"
+            }
+        }
+    }
+    
+    # need some directories to play with
+    let base_path = ($"tmp_(random uuid)" | path expand )
+
+    try {
+        mkdir $base_path ($base_path | path join "a") ($base_path | path join "b")
+        cd $base_path
+        use dirs.nu
+
+        myassert eq 1 ($env.DIRS_LIST | length) "list is just pwd after initialization"
+        myassert eq $base_path $env.DIRS_LIST.0 "list is just pwd after initialization"
+
+        dirs next
+        myassert eq $base_path $env.DIRS_LIST.0 "next wraps at end of list"
+
+        dirs prev
+        myassert eq $base_path $env.DIRS_LIST.0 "prev wraps at top of list"
+
+        dirs add ($base_path | path join "b") ($base_path | path join "a")
+        myassert eq ($base_path | path join "b") $env.PWD "add changes PWD to first added dir"
+        myassert eq 3 ($env.DIRS_LIST | length) "add in fact adds to list"
+        myassert eq ($base_path | path join "a") $env.DIRS_LIST.2 "add in fact adds to list"
+        
+        dirs next 2
+        myassert eq $base_path $env.PWD "next wraps at end of list"
+
+        dirs prev 1 
+        myassert eq ($base_path | path join "a") $env.PWD "prev wraps at start of list"
+
+        dirs drop
+        myassert eq 2 ($env.DIRS_LIST | length) "drop removes from list"    
+        myassert eq $base_path $env.PWD "drop changes PWD to next in list (after dropped element)"
+
+        let show_tab = (dirs show)
+        myassert eq $show_tab [[active path]; [true $base_path] [false ($base_path | path join "b")]] "show table contains expected information"
+    } catch { |error|
+        $error | debug
+        true
+    }
+
+    cd $base_path
+    cd ..
```

Comment:
you do not need to undo the `cd`s in a regular `def` function, 'cause the PWD does not change outside the scope of the function.

you might be familiar with subshells in `bash` where
```bash
# here you in some initial-place/
{
    cd somewhere
    # do something in somewhere/
}
# and you're back to initial-place/ without any `cd`ing back
```

Reply:
Is the localized CD true of changes within any closure (like a try/catch or if/then), or only within scope of function body?
I was also concerned that a *failing* test might leave the working directory indeterminate, which might make the directory cleanup at the end unreliable.  So I think I have to CD into the parent directory before I try to `rm -r` the temporary directories, since there is some risk I might still have working directory set to one of the directories I'm trying to delete.
And really, the extra CD is cheap insurance in a test, it's not mainline code...  I want to leave it as is.

Reply:
this disk directory might be not trivial to work with :thinking:

Reply:
I would agree with @bobhy here that we probably should play it defensive for now. It is not totally unprecedented that our semantics in this area have changed. And if we can keep the blast radius minimal for stuff like that we get a little peace of mind.

Reply:
but what about just putting all these directories inside `$nu.temp-path` instead? :thinking: 

that would solve the issue of cleaning them => they would be at next reboot for example and the amount of things being generated is minimal